src/Sylius/Bundle/CoreBundle/Provider/SymfonyTranslatorResourceProvider.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Provider;
+
+use Sylius\Bundle\ThemeBundle\Translation\Provider\Resource\TranslatorResourceProviderInterface;
+use Sylius\Bundle\ThemeBundle\Translation\Resource\TranslationResource;
+use Sylius\Bundle\ThemeBundle\Translation\Resource\TranslationResourceInterface;
+
+/** @internal */
+class SymfonyTranslatorResourceProvider implements TranslatorResourceProviderInterface
+{
+    private TranslatorResourceProviderInterface $decorated;
+
+    /** @var TranslationResourceInterface[] */
+    private array $resources = [];
+
+    private array $resourcesLocales = [];
+
+    private array $filepaths = [];
+
+    public function __construct(TranslatorResourceProviderInterface $decorated)
+    {
+        $this->decorated = $decorated;
+    }
+
+    public function getResources(): array
+    {
+        $this->initialize();
+
+        return $this->resources;
+
+    }
+
+    public function getResourcesLocales(): array
+    {
+
+        $this->initialize();
+
+        return $this->resourcesLocales;
+    }
+
+    /**
+     * 1. The function does not initialize if needed. It initializes everytime it is executed.
+     * 2. The function appends resources and locales each time it is executed, meaning that if I have 100 translation
+     * files and execute it
+     */
+    private function initialize(): void
```

Comment:
@SirDomin the phpdoc would be the thing you would like to put in the Description. Second, there is no fix ;p The SymfonyTranslatorResourceProvider is 1to1 with the ThemeBundle one.